### PR TITLE
500 - Internal Server Error

### LINKED ISSUE #20: 500 - Internal Server Error
If somethings goes wrong inside the server while handling a request giving an exception we can respond with an error message of type 500.  https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500

### PR DESCRIPTION
**Pull request by albindubech**
I think this resolves #20

### CODE REVIEW

**Comment by LordRekishi** [comment]
I think you got conflicts because of the changes made in merging Issue #40 earlier today. Maybe if you merge in the new main branch it could be solved?

**Comment by albindubech** [comment]
> I think you got conflicts because of the changes made in merging Issue #40 earlier today. Maybe if you merge in the new main branch it could be solved?

Whould I need to create a new Pull Request for this to be made?

**Comment by LordRekishi** [comment]
> > I think you got conflicts because of the changes made in merging Issue #40 earlier today. Maybe if you merge in the new main branch it could be solved?
> 
> Whould I need to create a new Pull Request for this to be made?

Exactly. Update the project through intellij, and then merge main into your branch.

**Comment by Philippevial** [comment]
> .

I think the pull-request updates when you push from the same branch again, so you don't need to create a new one!
(After updating the branch of course)

**Comment by albindubech** [comment]
This pull request should now be updated with the latest changes form main

**Comment by kappsegla** [comment]
The HttpResponse class holding response header seams to be a good idea. Maybe call the class HttpResponseStatusCodes or something similar? And keep the HttpResponse class available for a complete response from the server?

Hardcoding a contact adress to fungover.org in the code is a bad idea, because this server should be able to run on many different hostnames, not just ours. Suggestion to just return 500 Internal Server Error because protocol version depends on how our server is configured so something else in the code has to define that.

**Comment by albindubech** [comment]
> The HttpResponse class holding response header seams to be a good idea. Maybe call the class HttpResponseStatusCodes or something similar? And keep the HttpResponse class available for a complete response from the server?
> 
> Hardcoding a contact adress to fungover.org in the code is a bad idea, because this server should be able to run on many different hostnames, not just ours. Suggestion to just return 500 Internal Server Error because protocol version depends on how our server is configured so something else in the code has to define that.

Should I rename the class to "HttpResponseStatusCodes" and leave the "HTTpResponse" name available for later use then? Or did I misunderstand?

File: src/main/java/org/fungover/storm/client/HttpResponseStatusCodes.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.fungover.storm.client;
+
+public class HttpResponseStatusCodes {
+    static final String error500 = "HTTP/1.1 500 Internal Server Error\r\n\r\n";
```
Comment (kappsegla) [inline src/main/java/org/fungover/storm/client/HttpResponseStatusCodes.java:4]:
```suggestion
    static final String error500 = "500 Internal Server Error\r\n\";
```

File: src/main/java/org/fungover/storm/client/ClientHandler.java [hunk 2]
```diff
@@ -32,7 +33,11 @@ public void run() {
             out.close();
             clientSocket.close();
         } catch (IOException e) {
-            LOGGER.error(e.getMessage());
+            statusCode = new HttpResponseStatusCodes();
```
Comment (kappsegla) [inline src/main/java/org/fungover/storm/client/ClientHandler.java:36]: Could there be a check here in the future to see if the connection still is open and no data has ben transfered to the client yet. In that case try to send a response to the client containing the 500 error.
Maybe create a new issue for that?

**Comment by kappsegla** [comment]
> > The HttpResponse class holding response header seams to be a good idea. Maybe call the class HttpResponseStatusCodes or something similar? And keep the HttpResponse class available for a complete response from the server?
> > Hardcoding a contact adress to fungover.org in the code is a bad idea, because this server should be able to run on many different hostnames, not just ours. Suggestion to just return 500 Internal Server Error because protocol version depends on how our server is configured so something else in the code has to define that.
> 
> Should I rename the class to "HttpResponseStatusCodes" and leave the "HTTpResponse" name available for later use then? Or did I misunderstand?

Much better name like this.

File: src/main/java/org/fungover/storm/client/HttpResponseStatusCodes.java (hunk 1 above)
Comment (albindubech) [reply src/main/java/org/fungover/storm/client/HttpResponseStatusCodes.java]: > 

PR updated with the suggestion

**Comment by Vimbayinashe** [comment]
This looks good! It would be good to add tests for these changes as soon as #62 has received a second review.

**Comment by albindubech** [comment]
> This looks good! It would be good to add tests for these changes as soon as #62 has received a second review.

Good suggestion, will update with tests when #62 is completed.

**Comment by Vimbayinashe** [comment]
> > This looks good! It would be good to add tests for these changes as soon as #62 has received a second review.
> 
> Good suggestion, will update with tests when #62 is completed.

Pull request #62 is now marked as draft as it has become dependent on another issue. I will approve this pull request now.

**Review by Vimbayinashe — APPROVED** [review]

**Review by ErikssonF — APPROVED** [review]
Looks good!

**Pull request merged**